### PR TITLE
feat: Option to prevent propagating collision events from ShapeHitbox to _hitboxParent

### LINKED ISSUE #2590: Option to prevent propogating collision events from ShapeHitbox to _hitboxParent
# What could be improved
In some complex cases a component need to have several hitboxes and some of them could have very utility function. For example, a vechicle could have body hitbox to control collisions and side hitboxes to check availability to turn left or right. So, colliding with body hitbox means a collision with component itself, whereas colliding with side hitbox does not mean a real collision and should not be propogated to hitbox's parent. 

# Why should this be improved
Of course we can do these checks in 'on collsion' functions or handle utulity logics there, but this is not flexible if we want to separate logics to components, to use ECS approach and so on. 

# Any risks?
Just one option to ShapeHitbox, which could be completely ignored if not needed.

### PR DESCRIPTION
**Pull request by ASGAlex**
# Description

See #2590 

Added `triggersParentCollision` variable to `ShapeHitbox` class to prevent propogating `onCollision*` events to parent component

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Closes #2590

### CODE REVIEW

**Comment by ASGAlex** [comment]
Hmmm, any ideas, what is wrong with tests?....

**Comment by spydon** [comment]
> Hmmm, any ideas, what is wrong with tests?.... 

There has been something strange going on in the GitHub actions lately so that it stops mid-run, re-ran it now and all is green.

File: doc/flame/collision_detection.md [hunk 1]
```diff
@@ -138,6 +138,42 @@ initiate hitboxes and it is with the normal constructor where you define the hit
 a size and a position etc. The other way is to use the `relative` constructor which defines the
 hitbox in relation to the size of its intended parent.
 
+In some specific cases you might want to handle collisions only between hitboxes, without
+propagating 'onCollision*' events to hitbox's parent component. For example, a vehicle could have
```
Comment (spydon) [inline doc/flame/collision_detection.md:142]:
```suggestion
propagating `onCollision* events to the hitbox's parent component. For example, a vehicle could have a
```

File: doc/flame/collision_detection.md [hunk 2]
```diff
@@ -138,6 +138,42 @@ initiate hitboxes and it is with the normal constructor where you define the hit
...
+body hitbox to control collisions and side hitboxes to check availability to turn left or right.
```
Comment (spydon) [inline doc/flame/collision_detection.md:143]:
```suggestion
body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
```

File: doc/flame/collision_detection.md [hunk 3]
```diff
@@ -138,6 +138,42 @@ initiate hitboxes and it is with the normal constructor where you define the hit
...
+So, colliding with body hitbox means a collision with component itself, whereas colliding with side
```
Comment (spydon) [inline doc/flame/collision_detection.md:144]:
```suggestion
So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
```

File: packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart [hunk 4]
```diff
@@ -27,6 +27,10 @@ mixin ShapeHitbox on ShapeComponent implements Hitbox<ShapeHitbox> {
   /// added to the same parent.
   bool allowSiblingCollision = false;
 
+  /// Whether the hitbox collision with other hitbox should trigger
```
Comment (spydon) [inline packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart:30]:
```suggestion
  /// Whether hitbox collisions with other hitboxes should trigger the
```

File: packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart [hunk 5]
```diff
@@ -27,6 +27,10 @@ mixin ShapeHitbox on ShapeComponent implements Hitbox<ShapeHitbox> {
...
+  /// "onCollision" functions for hitbox's parent component.
```
Comment (spydon) [inline packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart:31]:
```suggestion
  /// "onCollision" functions for the hitbox's parent component.
```

**Review by spydon — COMMENTED** [review]
Just some grammar comments, the code itself looks good

File: doc/flame/collision_detection.md [hunk 6]
```diff
@@ -138,6 +138,42 @@ initiate hitboxes and it is with the normal constructor where you define the hit
 a size and a position etc. The other way is to use the `relative` constructor which defines the
 hitbox in relation to the size of its intended parent.
 
+In some specific cases you might want to handle collisions only between hitboxes, without
+propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could have a
+body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
+So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
+hitbox does not mean a real collision and should not be propagated to hitbox's parent.
+For this case you can set `triggersParentCollision` variable to `false`:
```
Comment (spydon) [inline doc/flame/collision_detection.md:146]:
```suggestion

In some specific cases you might want to handle collisions only between hitboxes, without
propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could have a
body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
hitbox does not mean a real collision and should not be propagated to hitbox's parent.
For this case you can set `triggersParentCollision` variable to `false`:
```

**Review by spydon — APPROVED** [review]

File: doc/flame/collision_detection.md [hunk 7]
```diff
@@ -138,6 +138,43 @@ initiate hitboxes and it is with the normal constructor where you define the hit
 a size and a position etc. The other way is to use the `relative` constructor which defines the
 hitbox in relation to the size of its intended parent.
 
+
+In some specific cases you might want to handle collisions only between hitboxes, without
+propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could have a
+body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
+So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
+hitbox does not mean a real collision and should not be propagated to hitbox's parent.
+For this case you can set `triggersParentCollision` variable to `false`:
```
Comment (spydon) [inline doc/flame/collision_detection.md:147]:
```suggestion
In some specific cases you might want to handle collisions only between hitboxes, without
propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could
have a body hitbox to control collisions and side hitboxes to check the possibility to turn left
or right.
So, colliding with a body hitbox means colliding with the component itself, whereas colliding with
a side hitbox does not mean a real collision and should not be propagated to hitbox's parent.
For this case you can set `triggersParentCollision` variable to `false`:
```

**Pull request merged**